### PR TITLE
Feature/json processor

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: src/Nest/Ingest/Processors/KeyValueProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	[JsonObject(MemberSerialization.OptIn)]
+	[JsonConverter(typeof(ProcessorJsonConverter<KeyValueProcessor>))]
+	public interface IKeyValueProcessor : IProcessor
```
Comment (russcam) [inline src/Nest/Ingest/Processors/KeyValueProcessor.cs]: File needs to be included in NEST

File: src/Nest/Ingest/ProcessorsDescriptor.cs [hunk 2]
```diff
@@ -106,6 +106,12 @@ public class ProcessorsDescriptor : DescriptorPromiseBase<ProcessorsDescriptor,
 		public ProcessorsDescriptor Uppercase<T>(Func<UppercaseProcessDescriptor<T>, IUppercaseProcessor> selector) where T : class  =>
 			Assign(a => a.AddIfNotNull(selector?.Invoke(new UppercaseProcessDescriptor<T>())));
 
+		/// <summary>
+		/// Converts a JSON string into a structured JSON object.
+		/// </summary>
+		public ProcessorsDescriptor Json<T>(Func<JsonProcessorDescriptor<T>, IJsonProcessor> selector) where T : class  =>
```
Comment (russcam) [inline src/Nest/Ingest/ProcessorsDescriptor.cs]: Also add method for `KeyValueProcessor`

File: src/Nest/Ingest/Processors/KeyValueProcessor.cs (hunk 1 above)
Comment (russcam) [inline src/Nest/Ingest/Processors/KeyValueProcessor.cs]: Should this be called `KVProcessor` to [align with the docs?](https://www.elastic.co/guide/en/elasticsearch/reference/5.x/kv-processor.html)

File: src/Nest/Ingest/PipelineJsonConverter.cs [hunk 3]
```diff
@@ -57,6 +57,9 @@ private List<IProcessor> GetProcessors(JToken jsonProcessors, JsonSerializer ser
 					case "foreach":
 						processors.Add(jsonProcessor.ToObject<ForeachProcessor>(serializer));
 						break;
+					case "json":
```
Comment (russcam) [inline src/Nest/Ingest/PipelineJsonConverter.cs]: Add deserialization for `KeyValueProcessor`

**Comment by Mpdreamz** [comment]
ported to `master`

**Pull request merged**